styles/scss/core/_variables.scss

```diff
@@ -320,7 +320,7 @@ $escaped-characters: (
 
 $enable-caret:                                true !default;
 $enable-rounded:                              true !default;
-$enable-shadows:                              false !default;
+$enable-shadows:                              true !default;
```

Comment:
For some reason, we have a problem with the display of shadows generated by the Bootstrap tool. 
I tried to include the flag above and earned)

Reply:
Oh I see what the issue is 😔. The problem with your solution is that this flag also enables box shadows for `openedx` theme which we shouldn't enable (because they are turned off in `master` branch). This flag is currently turned on in edX theme [here](https://github.com/edx/brand-edx.org/blob/master/paragon/_variables.scss#L251), which overrides the flag in Paragon and allows to display shadows in edX theme.

However, this flag does not work in `alpha` branch because we don't override SCSS variables anymore since we use CSS variables... And I don't think there is way to make these global flags work with CSS variables.

I *think* the only solution here is to deprecate this flag and remove usage of [box-shadow mixin](https://github.com/twbs/bootstrap/blob/cdcb503b88a6d3ddcc870022a740b996af0034d3/scss/mixins/_box-shadow.scss#L1C4-L1C4) from our codebase because it relies on this flag. Instead, box-shadows should only be controlled by CSS variables, e.g. for `Dropdown` component we should replace this line https://github.com/openedx/paragon/blob/cc022bb0b071101d0bc3a6392f0ec69f5b874e5d/src/Dropdown/dropdown-bootstrap.scss#L33

with just
```scss
box-shadow: var(--pgn-elevation-dropdown-box-shadow);
```

and set `--pgn-elevation-dropdown-box-shadow` variable to be `none` in Paragon but to be actual box-shadow value in edX brand theme.

We'll probably need to do the same for all other global SCSS flags...

@adamstankiewicz curious what you think here 🙂

Reply:
Yeah, it's one of the possible solutions. Looks interesting, thanks 💯

Reply:
> I think the only solution here is to deprecate this flag and remove usage of [box-shadow mixin](https://github.com/twbs/bootstrap/blob/cdcb503b88a6d3ddcc870022a740b996af0034d3/scss/mixins/_box-shadow.scss#L1C4-L1C4) from our codebase because it relies on this flag. Instead, box-shadows should only be controlled by CSS variables

I agree with what @viktorrusakov has mentioned, where the `Dropdown` box shadow for openedx theme should be `none` but for edX.org, it should be the actual value.

[clarification] Regarding deprecation, is there a way for us to legit deprecate these flags (i.e., have them still function as expected but warn that they will be soon removed). If we can't support the flag's functionality for consumers as is, then that feels like it is a breaking change that we may need to introduce now instead of a true deprecation? cc @viktorrusakov 

[suggestion/question] When replacing the usage of the SCSS mixin with the CSS variable (e.g., `box-shadow: var(--pgn-elevation-dropdown-box-shadow);`), I might also consider if/when it makes sense to use the `pgn-box-shadow` mixin in the internal Paragon code. Do we intend to keep that `pgn-box-shadow` SCSS mixin around for consumers to use with design tokens?

> We'll probably need to do the same for all other global SCSS flags...

@PKulkoRaccoonGang @monteri @viktorrusakov Do we have a good understanding of what all the other global SCSS flags are if we are considering deprecating/removing them, too?

Reply:
> Do we have a good understanding of what all the other global SCSS flags are if we are considering deprecating/removing them, too?

I don't think there should be any global SCSS flags anymore, SCSS shouldn't be able to control styles like this since design tokens are the single source of truth for styles now.  We should consider migrating some flags to the design tokens schema I think (e.g. to be able to disable all box shadows by just changing one global token).

I think these are the only ones we have to take care for, I don't remember seeing any other flags, but we'll have to make sure of that also.
https://github.com/openedx/paragon/blob/932eed0dcba4ec849dd825cbcd0473b9567c7cea/scss/core/_variables.scss#L321-L333

For what it's worth I don't think consumers use these flags much 😅. For example edX.org brand [overrides](https://github.com/edx/brand-edx.org/blob/master/paragon/_variables.scss#L251) only the shadows flag. Removing them would still be a breaking change for sure, I'm just saying that I don't think it's that big of a breaking change 🙂 

> Do we intend to keep that `pgn-box-shadow` SCSS mixin around for consumers to use with design tokens?

I think it's ok to keep them for now just to minimize amount of breaking changes. And I think they might still be useful to engineers so they could just write `@include pgn-box-shadow(1, "down")` instead of looking up CSS variable for that. I would probably suggest we change internal implementation of the mixin so it uses CSS variables directly instead of SCSS ones.

> Regarding deprecation, is there a way for us to legit deprecate these flags (i.e., have them still function as expected but warn that they will be soon removed).

We could do that, but the problem is that it's an SCSS flag, which means that if consumers want to keep using them, they need to do that through SCSS, i.e. keep importing `variables.scss` file from their brand package into MFEs. They won't be able to override these flags just by including CSS files into the HTML head. That's actually why this flag doesn't work in `alpha` branch - we no longer import SCSS files from edX.org brand in docs site, so this flag is not overridden anymore.

Apart from this I don't see any problems with deprecating these flags just by overriding Bootstrap's / Paragon's mixins with inclusion of deprecation message there (and maybe other minor tweaks).

Reply:
@adamstankiewicz @viktorrusakov thanks for the suggestions. Please look at the new changes! 👀

Reply:
There is only difference in variable is `$enable-box-shadows` that is `true` in the `brand-edx.org`. On one of the meeting we decided to remove variables completely BUT there are some dependencies on the React-Bootstrap side. So my current suggestion is that we will leave `scss flags` as is in the Paragon side so that styles relying on the React-Bootstrap would not break.

Reply:
@adamstankiewicz Regarding [the rest of the Bootstrap flags](https://github.com/openedx/paragon/blob/932eed0dcba4ec849dd825cbcd0473b9567c7cea/scss/core/_variables.scss#L321-L333) that we have in Paragon (except for `$enable-box-shadows`). At the moment, in order to completely transfer all flags to design tokens, we have several problems:
- [SCSS code blocks that are controlled by conditions](https://github.com/twbs/bootstrap/blob/v4.6.2/scss/_card.scss#L203-L236) (to solve this problem, may need to create multiple design tokens for each property)
- transferring Bootstrap files to the local `bootstrap-override` folder. This decision is inevitable because in the Paragon we are based on the SCSS of moisture that Bootstrap provides and for control from a “single source of truth” - the design of tokens, it is necessary to have full access to all Bootstrap files that use conditions that depend on the SCSS flags.

**Conclusion and proposals:** Having discussed with @monteri  and @viktorrusakov, we are inclined to believe that we need to abandon the partial use of Bootstrap and transfer it completely locally to the project.
This solution will open up the possibility for us to fully optimize styles for CSS variables and design tokens. Will provide a chance to refactor existing styles and remove unnecessary ones.

Reply:
This is not a blocker for this PR, so I'm going to merge it. We'll handle other SCSS flags in separate PRs.